packages/documentation-framework/hooks/useTheme.js

```diff
@@ -0,0 +1,127 @@
+import { useState, useEffect, useCallback } from 'react';
+
+const THEME_MODES = {
+  SYSTEM: 'system',
+  LIGHT: 'light',
+  DARK: 'dark'
+};
+
+const THEME_STORAGE_KEY = 'theme-preference';
+const DARK_MODE_CLASS = 'pf-v6-theme-dark';
+
+/**
+ * Custom hook for managing theme state with system preference detection
+ * @returns {Object} Theme state and controls
+ * @returns {string} themeMode - Current theme mode ('system'|'light'|'dark')
+ * @returns {Function} setThemeMode - Function to change theme mode
+ * @returns {string} resolvedTheme - The actual applied theme ('light'|'dark')
+ * @returns {Object} THEME_MODES - Available theme mode constants
+ */
+export const useTheme = () => {
+  const getStoredThemeMode = () => {
+    if (typeof window === 'undefined' || !window.localStorage) return null;
```

Comment:
This is fairly nitpicky and more to follow existing syntax, but rather than the inline return, just wrapping the curly braces around:

```suggestion
    if (typeof window === 'undefined' || !window.localStorage) {
    return null;  
  }
```

Which would apply to other returns in this hook as well, but again fairly nitpicky.